aws/httpServer.ts

```diff
@@ -40,82 +39,23 @@ export class HttpServer extends pulumi.ComponentResource implements cloud.HttpSe
             () => createLambdaEntryPoint(createRequestListener),
             { parent: this });
 
-        const lambdaOperation = createLambdaOperation(func.lambda);
-
-        const swagger = {
-            swagger: "2.0",
-            info: { title: name, version: "1.0" },
-            "x-amazon-apigateway-binary-media-types": [ "*/*" ],
-            "x-amazon-apigateway-gateway-responses": {
-                "MISSING_AUTHENTICATION_TOKEN": {
-                    "statusCode": 404,
-                    "responseTemplates": {
-                        "application/json": "{\"message\": \"404 Not found\" }",
-                    },
+        const api = new aws.apigateway.x.API(name, {
+            // Register two paths in the Swagger spec, for the root and for a catch all under the root
+            routes: [
+                {
+                    path: "/",
+                    method: "ANY",
+                    handler: func.lambda,
                 },
-                "ACCESS_DENIED": {
-                    "statusCode": 404,
-                    "responseTemplates": {
-                        "application/json": "{\"message\": \"404 Not found\" }",
-                    },
+                {
+                    path: "/{proxy+}",
+                    method: "ANY",
+                    handler: func.lambda,
                 },
```

Comment:
much simpler now.  All the core logic is in the pulumi/aws layer.  Now, all this layer needs to do is setup the proper `AWS->to->Node` lambda, and the setup a couple of simple swagger routes.